schneiderWallbox/integrationpluginschneiderwallbox.cpp

```diff
@@ -0,0 +1,546 @@
+    switch (state) {
+    case SchneiderWallboxModbusTcpConnection::CPWStateEvseNotAvailable:
+        thing->setStateValue("cpwState", "EVSE not available");
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateEvseAvailable:
+        thing->setStateValue("cpwState", "EVSE available");
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStatePlugDetected:
+        thing->setStateValue("cpwState", "Plug detected");
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateEvConnected:
+        thing->setStateValue("cpwState", "EV connected");
+        isPluggedIn = true;
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateEvConnected2:
+        thing->setStateValue("cpwState", "EV connected 2");
+        isPluggedIn = true;
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateEvConnectedVentilationRequired:
+        thing->setStateValue("cpwState", "EV connected ventilation required");
+        isPluggedIn = true;
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateEvseReady:
+        thing->setStateValue("cpwState", "EVSE ready");
+        isPluggedIn = true;
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateEvReady:
+        thing->setStateValue("cpwState", "EV ready");
+        isPluggedIn = true;
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateCharging:
+        thing->setStateValue("cpwState", "Charging");
+        isPluggedIn = true;
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateEvReadyVentilationRequired:
+        thing->setStateValue("cpwState", "EV ready ventilation required");
+        isPluggedIn = true;
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateChargingVentilationRequired:
+        thing->setStateValue("cpwState", "Charging ventilation required");
+        isPluggedIn = true;
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateStopCharging:
+        thing->setStateValue("cpwState", "Stop charging");
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateAlarm:
+        thing->setStateValue("cpwState", "Alarm");
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateShortcut:
+        thing->setStateValue("cpwState", "Short circuit");
+        break;
+    case SchneiderWallboxModbusTcpConnection::CPWStateDigitalComByEvseState:
+        thing->setStateValue("cpwState", "Digital com by EVSE state");
+        break;
+    default:
+        thing->setStateValue("cpwState", "EVSE not available");
+    }
+    thing->setStateValue("pluggedIn", isPluggedIn);
+}
+
+void IntegrationPluginSchneiderWallbox::setLastChargeStatus(Thing *thing, SchneiderWallboxModbusTcpConnection::LastChargeStatus status)
+{
+    switch (status) {
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusCircuitBreakerEnabled:
```

Comment:
No need to change this, just as a hint for the next time. I normally do something like this instead:
```
QHash<Enum, QString> map = {
  { EnumValue1, "String value for enum value 1" },
  { EnumValue2, "String value for enum value 2" },
};
thing->setStateValue("lastChargeStatus", map.value(status));
```
IMO nicer to read, but again, nothing wrong your code here.